uaclient/cli.py

```diff
@@ -1223,40 +1223,40 @@ def action_help(args, *, cfg):
     return 0
 
 
-def setup_logging(console_level, log_level, log_file=None):
+def setup_logging(console_level, log_level, log_file=None, logger=None):
     """Setup console logging and debug logging to log_file"""
     if log_file is None:
-        log_file = config.CONFIG_DEFAULTS["log_file"]
+        cfg = config.UAConfig()
+        log_file = cfg.log_file
     console_formatter = util.LogFormatter()
     log_formatter = logging.Formatter(DEFAULT_LOG_FORMAT)
-    root = logging.getLogger()
-    root.setLevel(log_level)
+    if logger is None:
+        # Then we configure the root logger
+        logger = logging.getLogger()
+    logger.setLevel(log_level)
+
+    # Our handlers are the best
+    logger.handlers = []
```

Comment:
I think this is reasonable for use to clear out existing loggers on each call to setup_logging. That said, I think it is worth a more descriptive  comment about "# clearing all handlers so any previous handlers for this logger are replaced"

Reply:
LOL this comment was not meant to be there at all, it was just part of the development thinking process :eyes: 
Will update